### PR TITLE
Added progress updater in UploadService to show upload progress in no…

### PR DESCRIPTION
**Pull request by ashishkumar468**
* Show notification progress while uploading images

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/upload/UploadClient.java [hunk 1]
```diff
@@ -31,11 +33,15 @@ public UploadClient(UploadInterface uploadInterface, @Named(NAMED_COMMONS_CSRF)
...
 
-    Observable<UploadResult> uploadFileToStash(Context context, String filename, File file) {
-        RequestBody requestFile = RequestBody
+    Observable<UploadResult> uploadFileToStash(Context context, String filename, File file,
+            NotificationUpdateProgressListener notificationUpdater) {
+        RequestBody requestBody = RequestBody
                 .create(MediaType.parse(FileUtils.getMimeType(context, Uri.parse(file.getPath()))), file);
 
-        MultipartBody.Part filePart = MultipartBody.Part.createFormData("file", filename, requestFile);
+        CountingRequestBody countingRequestBody=new CountingRequestBody(requestBody,
```
Comment (maskaravivek) [inline app/src/main/java/fr/free/nrw/commons/upload/UploadClient.java]: fix formatting

File: app/src/main/java/fr/free/nrw/commons/upload/UploadService.java [hunk 2]
```diff
@@ -202,23 +202,13 @@ public int onStartCommand(Intent intent, int flags, int startId) {
...
-        try {
-            file1 = new File(localUri.getPath());
-            fileInputStream = new FileInputStream(file1);
-        } catch (FileNotFoundException e) {
-            Timber.d("File not found");
-            Toast fileNotFound = Toast.makeText(this, R.string.upload_failed, Toast.LENGTH_LONG);
-            fileNotFound.show();
-            return;
-        }
+        File file1 =new File(localUri.getPath());
```
Comment (maskaravivek) [inline app/src/main/java/fr/free/nrw/commons/upload/UploadService.java]: - formatting.
- use a more meaningful variable name.

File: app/src/main/java/fr/free/nrw/commons/upload/UploadService.java [hunk 3]
```diff
@@ -250,7 +240,7 @@ private void uploadContribution(Contribution contribution) {
                     }
                 })
                 .flatMap(uploadStash -> {
-                    notificationManager.cancel(NOTIFICATION_UPLOAD_IN_PROGRESS);
+                    notificationManager.cancel(notificationTag,NOTIFICATION_UPLOAD_IN_PROGRESS);
```
Comment (maskaravivek) [inline app/src/main/java/fr/free/nrw/commons/upload/UploadService.java]: formatting

File: app/src/main/java/fr/free/nrw/commons/upload/UploadService.java [hunk 4]
```diff
@@ -293,7 +283,7 @@ private void uploadContribution(Contribution contribution) {
                     }
                 }, throwable -> {
                     Timber.w(throwable, "Exception during upload");
-                    notificationManager.cancel(NOTIFICATION_UPLOAD_IN_PROGRESS);
+                    notificationManager.cancel(notificationTag,NOTIFICATION_UPLOAD_IN_PROGRESS);
```
Comment (maskaravivek) [inline app/src/main/java/fr/free/nrw/commons/upload/UploadService.java]: formatting

File: app/src/main/java/fr/free/nrw/commons/upload/CountingRequestBody.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+import okio.Sink;
+
+/**
+ * Decorates an OkHttp request body to count the number of bytes written when writing it. Can
+ * decorate any request body, but is most useful for tracking the upload progress of large multipart
+ * requests.
+ *
+ * @author Leo Nikkilä
+ */
+public class CountingRequestBody extends RequestBody {
```
Comment (maskaravivek) [inline app/src/main/java/fr/free/nrw/commons/upload/CountingRequestBody.java]: Add java docs for methods added in this class. :)

**Comment by ashishkumar468** [comment]
Thanks for the comments @maskaravivek. I have pushed the said changes

**Comment by misaochan** [comment]
Tested on Nexus S API 27 emulator, progress bar in notification works well for me. Thanks @ashishkumar468 . :)

**Pull request merged**